src/main/java/org/graylog/integrations/dataadpaters/GreyNoiseDataAdapter.java

```diff
@@ -0,0 +1,202 @@
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.graylog2.plugin.lookup.LookupCachePurge;
+import org.graylog2.plugin.lookup.LookupDataAdapter;
+import org.graylog2.plugin.lookup.LookupDataAdapterConfiguration;
+import org.graylog2.plugin.lookup.LookupResult;
+import org.graylog2.security.encryption.EncryptedValue;
+import org.graylog2.security.encryption.EncryptedValueService;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotEmpty;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class GreyNoiseDataAdapter extends LookupDataAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(GreyNoiseDataAdapter.class);
+    public static final String NAME = "GreyNoise";
+    static final String GREYNOISE_IPQC_ENDPOINT = "https://api.greynoise.io/v2/noise/quick/";
+
+    private final EncryptedValueService encryptedValueService;
+    private final Config config;
+    private OkHttpClient okHttpClient;
+
+    @Inject
+    public GreyNoiseDataAdapter(@Assisted("id") String id,
+                                @Assisted("name") String name,
+                                @Assisted LookupDataAdapterConfiguration config,
+                                MetricRegistry metricRegistry,
+                                EncryptedValueService encryptedValueService) {
+        super(id, name, config, metricRegistry);
+        this.config = (Config) config;
+        this.encryptedValueService = encryptedValueService;
+    }
+
+    @Override
+    public void doStart() throws Exception {
+
+        okHttpClient = new OkHttpClient().newBuilder()
+                                         .build();
+
+    }
+
+    @Override
+    public void doStop() throws Exception {
+
+    }
+
+    @Override
+    public Duration refreshInterval() {
+        return Duration.ZERO;
+    }
+
+    @Override
+    protected void doRefresh(LookupCachePurge cachePurge) throws Exception {
+
+    }
+
+    @Override
+    protected LookupResult doGet(Object keyObject) {
+        try {
+
+            Request request = new Request.Builder()
+                    .url(GREYNOISE_IPQC_ENDPOINT + keyObject.toString())
+                    .method("GET", null)
+                    .addHeader("Accept", "application/json")
+                    .addHeader("key", encryptedValueService.decrypt(config.apiToken()))
+                    .build();
+
+            return parseResponse(okHttpClient.newCall(request).execute());
+
+        } catch (Exception e) {
+            LOG.error("An error occurred while retrieving lookup result [{}]", e.toString());
+            return LookupResult.withError();
+        }
+    }
+
+    public static LookupResult parseResponse(Response response) throws IOException {
```

Comment:
This doesn't really need to be public.  You can remove the `public` modifier to leave it package private and add a `@VisibleForTesting` annotation to explain why it's not `private`

Also, does it need to be static or is that just to make the test easier to write?

Reply:
to make it easier to write the test

Reply:
Ok, I don't think there's any danger to keeping it `static` since it doesn't interact with anything outside of its own scope.  But I do think it's worth dropping the `public` modifier. The JUnit shouldn't have any problem accessing the method since they're in the same package.

```suggestion
    static LookupResult parseResponse(Response response) throws IOException {
```